### ISSUE TITLE
New no-useless-override rule

### ISSUE THREAD

**Issue by manu-st** [palantir/tslint#3423]
### Feature Request

- __TSLint version__: 5.8
- __TypeScript version__: 2.5.3
- __Running TSLint via__: CLI / VSCode

#### TypeScript code being linted

```ts

   /* tslint:disable-next-line:prefer-function-over-method */
   foo ()
   {
      return this;
   }

```

with `tslint.json` configuration:

```json
{
   "rules": {
       "prefer-function-over-method": true
   }
}
```

#### Wanted Behavior
It would be great to have a new rule "no-useless-override" which triggers a warning or error when you use an override to disable a rule. Very often you had an override for a special case but through various refactoring this override is not needed anymore and the less override the better has you do not want to add code in the future that won't trigger the linter warning/error because you forgot to remove the useless rule.

Having a rule that checks for non-used override would be great.

#### Actual behavior
No warnings

#### Expected behavior
```
ERROR in C:/Dev/components/data/Navigable.ts
(70,4): Override "prefer-function-over-method" is not needed. Remove the local override.
```

**Comment by ajafff** [comment]
Duplicate of #1330?

**Comment by manu-st** [comment]
That seems to be the case.
